app/controllers/api/storage.php

```diff
@@ -1072,6 +1072,10 @@
 
         $decompressionTime = \microtime(true) - $startTime - $downloadTime - $decryptionTime;
 
+        if (empty($source)) {
+            $source = $deviceForFiles->read($path);
+        }
+
```

Comment:
A bit scared to do storage API change due to sites screenshot bug.
I would guess this scenario can never happen, otherwise we would have already fixed it here - it must be sites specific problem. So I would guess sites just dont create file document properly.

Please look into Builds.php (worker), and see how screenshot document is created. My guess would be, that there we dont set 'source' parameter. Compare that to `createFile` endpoint, here in storage.php, to see what expected value is. With that knowledge you should be able to have fix be done in Builds.php worker, and so we can then remove this fix in getFilePreview.

^ If this works, I would prefer that. if it doesn't, let's keep this fix, and I will ask other leads for review, to make sure this is safe to do.